rust/system/src/execution/orchestrator.rs

```diff
@@ -72,16 +71,25 @@ pub trait Orchestrator: Debug + Send + Sized + 'static {
         }
     }
 
+    /// Terminate the orchestrator with a result
+    async fn terminate_with_result(
+        &mut self,
+        res: Result<Self::Output, Self::Error>,
+        ctx: &ComponentContext<Self>,
+    ) {
+        self.default_terminate_with_result(res, ctx).await
+    }
+
     /// Terminate the orchestrator if the result is an error. Returns the output if any.
-    fn ok_or_terminate<O, E: Into<Self::Error>>(
+    async fn ok_or_terminate<O: Send, E: Into<Self::Error> + Send>(
```

Comment:
I don't understand why we are introducing Send bounds on O and then also making this async